src/OpenSpaceToolkit/Astrodynamics/Trajectory/Orbit/Models/Kepler/COE.cpp

```diff
@@ -483,7 +483,7 @@ COE COE::Cartesian(const COE::CartesianState& aCartesianState, const Derived& aG
 
         aop_rad = 0.0;
 
-        nu_rad = std::acos((nodeVector.dot(positionVector)) / (node * position));
+        nu_rad = std::acos(std::clamp((double)(nodeVector.dot(positionVector) / (node * position)), -1.0, 1.0));
```

Comment:
Worth adding a test for this case? Seems like an important functional part of the code.